app/components/category-item.js

```diff
@@ -15,4 +18,37 @@ export default Ember.Component.extend({
     let userCategories = this.get('userCategories');
     return userCategories.hasCategory(category);
   }),
+
+  actions: {
+    addCategory(category) {
+      this.set('isLoading', true);
+      this.addCategory(category).catch(() => {
```

Comment:
This is the beautiful thing about the closure actions. We're passing down `addCategory` into the component, and that's a function defined up top with `action (action addCategory)`, which itself is defined at the controller level.